WiseUnpacker/SabreTools.Serialization.Deserializers/WiseSectionHeader.cs

```diff
@@ -87,39 +87,193 @@ public class WiseSectionHeader : BaseBinaryDeserializer<SectionHeader>
             //Seek back to the beginning of the section
             data.Seek(initialOffset, 0);
 
-            header.UnknownValue0 = data.ReadUInt32LittleEndian();
+            header.UnknownDataSize = data.ReadUInt32LittleEndian();
             header.SecondExecutableFileEntryLength = data.ReadUInt32LittleEndian();
             header.UnknownValue2 = data.ReadUInt32LittleEndian();
             header.UnknownValue3 = data.ReadUInt32LittleEndian();
             header.UnknownValue4 = data.ReadUInt32LittleEndian();
             header.FirstExecutableFileEntryLength = data.ReadUInt32LittleEndian();
             header.MsiFileEntryLength = data.ReadUInt32LittleEndian();
-            if (headerLength == 6)
-                return header;
+            if (headerLength != 6)
+            {
+                header.UnknownValue7 = data.ReadUInt32LittleEndian();
+                header.UnknownValue8 = data.ReadUInt32LittleEndian();
+                if (headerLength != 8)
+                {
+                    header.ThirdExecutableFileEntryLength = data.ReadUInt32LittleEndian();
+                    header.UnknownValue10 = data.ReadUInt32LittleEndian();
+                    header.UnknownValue11 = data.ReadUInt32LittleEndian();
+                    header.UnknownValue12 = data.ReadUInt32LittleEndian();
+                    header.UnknownValue13 = data.ReadUInt32LittleEndian();
+                    header.UnknownValue14 = data.ReadUInt32LittleEndian();
+                    header.UnknownValue15 = data.ReadUInt32LittleEndian();
+                    header.UnknownValue16 = data.ReadUInt32LittleEndian();
+                    header.UnknownValue17 = data.ReadUInt32LittleEndian();
+                    if (headerLength != 17)
+                    {
+                        header.UnknownValue18 = data.ReadUInt32LittleEndian();
+                    }
+                }
+            }
+            
 
-            header.UnknownValue7 = data.ReadUInt32LittleEndian();
-            header.UnknownValue8 = data.ReadUInt32LittleEndian();
-            if (headerLength == 8)
-                return header;
+            // Parse strings
+            // TODO: Count size of string section for later size verification
+            
+            data.Seek(initialOffset + localWisOffset, 0);
+            header.TmpString = data.ReadNullTerminatedString(Encoding.ASCII); // read .TMP string
+            header.GuidString = data.ReadNullTerminatedString(Encoding.ASCII); // read GUID string
```

Comment:
Please remove the comments after. It's apparent what is being read from the variable names.

Reply:
Should be fixed.